svm-decode/src/deserialize.rs

```diff
@@ -0,0 +1,288 @@
+            }
+            let value = i32::from_le_bytes(data[..4].try_into().unwrap());
+            Ok((DynValue::I32(value), &data[4..]))
+        }
+        DynType::I64 => {
+            if data.len() < 8 {
+                return Err(anyhow!(
+                    "Not enough data for i64: expected 8 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = i64::from_le_bytes(data[..8].try_into().unwrap());
+            Ok((DynValue::I64(value), &data[8..]))
+        }
+        DynType::I128 => {
+            if data.len() < 16 {
+                return Err(anyhow!(
+                    "Not enough data for i128: expected 16 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = i128::from_le_bytes(data[..16].try_into().unwrap());
+            Ok((DynValue::I128(value), &data[16..]))
+        }
+        DynType::U8 => {
+            if data.is_empty() {
+                return Err(anyhow!("Not enough data for u8: expected 1 byte, got 0"));
+            }
+            let value = data[0];
+            Ok((DynValue::U8(value), &data[1..]))
+        }
+        DynType::U16 => {
+            if data.len() < 2 {
+                return Err(anyhow!(
+                    "Not enough data for u16: expected 2 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = u16::from_le_bytes(data[..2].try_into().unwrap());
+            Ok((DynValue::U16(value), &data[2..]))
+        }
+        DynType::U32 => {
+            if data.len() < 4 {
+                return Err(anyhow!(
+                    "Not enough data for u32: expected 4 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = u32::from_le_bytes(data[..4].try_into().unwrap());
+            Ok((DynValue::U32(value), &data[4..]))
+        }
+        DynType::U64 => {
+            if data.len() < 8 {
+                return Err(anyhow!(
+                    "Not enough data for u64: expected 8 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = u64::from_le_bytes(data[..8].try_into().unwrap());
+            Ok((DynValue::U64(value), &data[8..]))
+        }
+        DynType::U128 => {
+            if data.len() < 16 {
+                return Err(anyhow!(
+                    "Not enough data for u128: expected 16 bytes, got {}",
+                    data.len()
+                ));
+            }
+            let value = u128::from_le_bytes(data[..16].try_into().unwrap());
+            Ok((DynValue::U128(value), &data[16..]))
+        }
+        DynType::Bool => {
+            if data.is_empty() {
+                return Err(anyhow!("Not enough data for bool: expected 1 byte, got 0"));
+            }
+            let value = data[0] != 0;
+            Ok((DynValue::Bool(value), &data[1..]))
+        }
+        DynType::Pubkey => {
```

Comment:
Can have `DynType::FixedArray(num_elems: u8, inner_type: Box<DynType>)` and express Pubkey as `DynType::FixedArray(32, Box(DynType::U8))`

Reply:
This corresponds to `Fixed sized arrays` in borsh spec here https://borsh.io/